src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -901,4 +902,144 @@ final case class EServicesApiServiceImpl(
+        case _                                            => Future.unit
+      }
+
+    def createZip(
+      folderName: String,
+      docs: Seq[(Array[Byte], String)],
+      interfaceFile: (Array[Byte], String),
+      config: Json
+    ): Array[Byte] = {
+      val bos    = new ByteArrayOutputStream()
+      val zipOut = new ZipOutputStream(bos)
+
+      docs.foreach { case (data, entryName) =>
+        zipOut.putNextEntry(new ZipEntry(s"$folderName/documents/$entryName"))
+        zipOut.write(data)
+        zipOut.closeEntry()
+      }
+
+      val (interfaceData, interfaceFileName) = interfaceFile
+      zipOut.putNextEntry(new ZipEntry(s"$folderName/$interfaceFileName"))
+      zipOut.write(interfaceData)
+      zipOut.closeEntry()
+
+      zipOut.putNextEntry(new ZipEntry(s"$folderName/configuration.json"))
+      zipOut.write(config.toString.getBytes("UTF-8"))
+      zipOut.closeEntry()
+
+      zipOut.close()
+      bos.toByteArray
+    }
+
+    val result: Future[(String, HttpEntity.Strict)] = for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      eServiceUuid   <- eserviceId.toFutureUUID
+      descriptorUuid <- descriptorId.toFutureUUID
+      eService       <- catalogProcessService.getEServiceById(eServiceUuid)
+      _              <- assertRequesterAllowed(eService.producerId)(organizationId)
+      _              <- checkRiskAnalysisForExport(eService)
+      descriptor     <- eService.descriptors
+        .find(_.id === descriptorUuid)
+        .toFuture(EServiceDescriptorNotFound(eService.id.toString, descriptorId))
+      _              <- verifyExportEligibility(descriptor)
+      interface      <- descriptor.interface.toFuture(MissingInterface(eServiceUuid, descriptorUuid))
+      interfaceFile  <- fileManager
+        .get(ApplicationConfiguration.eServiceDocumentsContainer)(interface.path)
+        .map(_.toByteArray)
+      _              <- FileManagerUtils
+        .verify(interfaceFile, interface.name, eService, isInterface = true)
+        .toFuture
+      docFiles       <- descriptor.docs.traverse(doc =>
+        fileManager
+          .get(ApplicationConfiguration.eServiceDocumentsContainer)(doc.path)
+          .map(filestream => (filestream.toByteArray, doc.name))
+      )
+      config     = extractConfig(eService, descriptor)
+      folderName = s"${eService.name}_V${}${descriptor.version}"
+      zipFile    = createZip(folderName, docFiles, (interfaceFile, interface.name), config)
+    } yield (s"$folderName.zip", HttpEntity(ContentType(MediaTypes.`application/octet-stream`), zipFile))
+
+    onComplete(result) {
+      val headers: List[HttpHeader] = headersFromContext()
+      handleError(s"Error exporting eservice $eserviceId with descriptor $descriptorId", headers) orElse {
+        case Success((filename, resource)) =>
+          complete(
+            StatusCodes.OK,
+            headers ++ Seq(`Content-Disposition`(attachment, Map("filename" -> filename))),
```

Comment:
the file download has the same size limitation of the file upload and we should use the presigned url approach.
In this case the steps could be
1. zip creation
2. zip upload to S3
3. GET presigned url generation
4. return the presigned url for the download